### PR TITLE
Sidecar module

### LINKED ISSUE #10: finish vpc setup
fix the resources setup in the root directory to bring up vpc,etc and pass the values into the modules [here](https://github.com/greymatter-io/terraform-greymatter-ecs/blob/master/greymatter/main.tf)

when this is done, remove the provider from that file.

### LINKED ISSUE #12: consider breaking out the sidecar into its own module
the work that would be needed for this would be:

vars:
- name
- port
- control dns

this could be a source-able module in itself, and would allow for the other modules as the come up - data, sense, etc to grab the sidecar without having to duplicate it for every service

### PR DESCRIPTION
**Pull request by zoemccormick**
I based this off branch `finish_vpc` so this includes the changes from pr #14 with updated ami

closes #10 
closes #12 

It moves the sidecar resources into its own module so we can repeat it for services with a sidecar. I added edge to the greymatter `main.tf` to show this

### CODE REVIEW

File: greymatter/modules/sidecar/secrets.tf [hunk 1]
```diff
@@ -0,0 +1,12 @@
+//resource "aws_ssm_parameter" "access_key" {
```
Comment (sofiashalotenko) [inline greymatter/modules/sidecar/secrets.tf]: Is this file necessary?

File: greymatter/variables.tf [hunk 2]
```diff
@@ -30,14 +12,10 @@ variable "key_pair_name" {
   default = "enter-ecs"
```
Comment (sofiashalotenko) [inline greymatter/variables.tf]: This probably isn't a great default...it just happens to be the key pair name I have, I think it slipped through one of my PRs.

File: greymatter/modules/sidecar/secrets.tf (hunk 1 above)
Comment (zoemccormick) [reply greymatter/modules/sidecar/secrets.tf]: no not right now  that was from an attempt at #11  ill delete

File: greymatter/modules/sidecar/tasks.tf [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+locals {
+  sidecar_container = <<DEFINITION
+[
+        {
+        "memoryReservation": 128,
+        "name": "${var.name}-sidecar",
+        "logConfiguration": {
+            "logDriver": "awslogs",
+            "options": {
+                "awslogs-group": "openjobs",
```
Comment (sofiashalotenko) [inline greymatter/modules/sidecar/tasks.tf]: I don't think there's terraform code that creates this openjobs cloudwatch log group.  Maybe it should be added somewhere in the infrastructure module?

**Comment by sofiashalotenko** [comment]
Control is failing for me with `ecs: could not retrieve ECS clusters: UnrecognizedClientException: The security token included in the request is invalid.`

File: greymatter/modules/sidecar/tasks.tf (hunk 3 above)
Comment (zoemccormick) [reply greymatter/modules/sidecar/tasks.tf]: im not sure what you mean by this? I dont have any experience using cloudwatch really, but the log group openjobs does appear for me when i run this and the logs for control, control-api, control-api sidecar and edge do appear

**Comment by zoemccormick** [comment]
@sofiashalotenko  Try changing [this](https://github.com/greymatter-io/terraform-greymatter-ecs/blob/32415c3a3f92c2e75e64dd6e8b0b2003924b64ab/main.tf#L44) to `ref=sidecar_module` in your local branch and terraform init again, this could be the problem

File: greymatter/modules/sidecar/tasks.tf (hunk 3 above)
Comment (sofiashalotenko) [reply greymatter/modules/sidecar/tasks.tf]: So I had to manually create a log group called "openjobs" in cloudwatch before this worked for me.  I think you maybe had openjobs created when you tried running my old ansible scripts (my best guess).  Anyway, if you go to `cloudwatch -> logs -> log groups` that's where it lives.  A quick resource like https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_log_group should create it.

**Review by sofiashalotenko — APPROVED** [review]
lgtm
